src/FSharp.Data.GraphQL.Server/ReflectionHelper.fs

```diff
@@ -140,6 +144,50 @@ module internal ReflectionHelper =
 
     let isPrameterMandatory = not << isParameterOptional
 
+    let unwrapOptions (ty : Type) =
+        if ty.FullName.StartsWith OptionTypeName || ty.FullName.StartsWith ValueOptionTypeName then
+            ty.GetGenericArguments().[0]
+        else ty
+
+    let isAssignableWithUnwrap (from: Type) (``to``: Type) =
+
+        let checkCollections (from: Type) (``to``: Type) =
+            if
+                // TODO: Implement support of other types of collections using collection initializers
+                (``to``.FullName.StartsWith ListTypeName || ``to``.FullName.StartsWith ArrayTypeName)
+                && (from.IsGenericType
+                    && from.GenericTypeArguments[0].IsAssignableTo(``to``.GenericTypeArguments[0])
+                    && from.GetInterfaces()
+                       |> Array.exists (
+                        fun i -> i.FullName.StartsWith IEnumerableGenericTypeName
+                                    || i.FullName = IEnumerableTypeName
+                       )
+                    )
+
+            then
+                let fromType = from.GetGenericArguments()[0]
+                let toType = ``to``.GetGenericArguments()[0]
+                fromType.IsAssignableTo toType
+            else
+                false
+
+        let from =
+            if from.FullName.StartsWith OptionTypeName || from.FullName.StartsWith ValueOptionTypeName then
+                from.GetGenericArguments()[0]
+            else from
+        let ``to`` =
+            if ``to``.FullName.StartsWith OptionTypeName || ``to``.FullName.StartsWith ValueOptionTypeName then
+                ``to``.GetGenericArguments()[0]
+            else ``to``
+
+        let result = from.IsAssignableTo ``to`` || checkCollections from ``to``
+        if result then result
+        else
+            if from.FullName.StartsWith OptionTypeName || from.FullName.StartsWith ValueOptionTypeName then
+                let from = from.GetGenericArguments()[0]
+                from.IsAssignableTo ``to`` || checkCollections from ``to``
+            else result
```

Comment:
It is hard to read

Reply:
ok, but did you also read the part:
 > also notice how the else result in your code is superfluous, since result is always false at that point.
 
 ?